tests/services/test_facets.py

```diff
@@ -10,68 +10,152 @@
         },
-    ]
-
-
-@pytest.fixture(scope='module')
-def example_records(database, identity, service, example_data):
-    """Create some example records."""
-    service.create_type(identity, 'languages', 'lng')
-    records = []
-    for data in example_data:
-        records.append(service.create(identity, data))
+        "title": {
+            "en": "Root One Sub One"
+        },
+        "type": "nested_vocab"
+    }))
+    vocabs.append(current_service.create(system_identity, {
+        "id": "rootone-subtwo",
+        "props": {
+            "type": "rootone-subtwo",
+            "type_name": "rootone-subtwo",
+        },
+        "title": {
+            "en": "Root One Sub Two"
+        },
+        "type": "nested_vocab"
+    }))
+    vocabs.append(current_service.create(system_identity, {
+        "id": "rootone-subone-subone",
+        "props": {
+            "type": "rootone-subone-subone",
+            "type_name": "rootone-subone-subone",
+        },
+        "title": {
+            "en": "Root One Sub Sub One"
+        },
+        "type": "nested_vocab"
+    }))
+    vocabs.append(current_service.create(system_identity, {
+        "id": "roottwo",
+        "props": {
+            "type": "roottwo",
+            "type_name": "roottwo",
+        },
+        "title": {
+            "en": "Root Two"
+        },
+        "type": "nested_vocab"
+    }))
 
     Vocabulary.index.refresh()
-    return records
-
 
-def _assert_vocabs(labels_instance):
-    labels = labels_instance(["eng", "spa"])
+    return vocabs
 
-    assert labels["eng"] == "English"
-    assert labels["spa"] == "Spanish"
 
+@pytest.fixture(scope="module")
+def nested_buckets():
+    class DummyData:
+        """Dummy data with buckets."""
+        buckets = [
+            LabellingBucket("rootone", 2),
+            LabellingBucket("rootone-subone", 1),
+            LabellingBucket("rootone-subone-subone", 4),
+            LabellingBucket("rootone-subtwo", 3),
+            LabellingBucket("roottwo", 5)
+        ]
 
-def test_vocabulary_label_cache(example_records):
-    # Uses example record to have two languages created on the vocabulary
-    labels_instance = VocabularyLabels(vocabulary="languages", cache=True)
-    _assert_vocabs(labels_instance)
+    return DummyData()
 
 
-def test_vocabulary_label_no_cache(example_records):
-    # Uses example record to have two languages created on the vocabulary
-    labels_instance = VocabularyLabels(vocabulary="languages", cache=False)
-    _assert_vocabs(labels_instance)
-
-
-def test_vocabulary_label_not_found(example_records):
-    # Uses example record to have two languages created on the vocabulary
-    labels_instance = VocabularyLabels(vocabulary="languages", cache=True)
-    labels = labels_instance(["eng", "glg"])
-
-    assert labels["eng"] == "English"
-    assert not labels.get("glg")  # search wont fail
+@pytest.fixture(scope="module")
+def nested_item_labels(app, nested_vocabulary):
```

Comment:
Just IMO: I would have added this to the test and not as a fixture if not re-usable.

Reply:
That's true, it just seemed cleaner with the values in fixtures. But changed :)